packages/lib-utils/src/hooks/useWorkspace.ts

```diff
@@ -18,34 +20,31 @@ import { getActiveWorkspace, setActiveWorkspace } from '../k8s/k8s-utils';
  * ```
  */
 export const useWorkspace = () => {
+  const { subscribe, unsubscribe, getState, setWorkspaceContext } = useContext(WorkspaceContext);
+  const [, forceRender] = useReducer((count) => count + 1, 0);
```

Comment:
@Hyperkid123 I'm a bit worried this is an anti-pattern WRT hooks in React. See their doc here: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate

> Normally, you shouldn’t mutate local state in React. However, as an escape hatch, you can use an incrementing counter to force a re-render even if the state has not changed
>
> Try to avoid this pattern if possible.

While this will work, it breaks the flow of updates and immutable data. Can we use something else to cause a re-render? Maybe some ID associated with the workspace?

Reply:
The issue is you want a mutable data structure (localStorage) to trigger react render cycle that depends purely on immutable data. So you have to force it. If you take a deep dive into some state management libraries that are framework agnostic, you will find that their react bindings do something very similar if not exactly the same thing.

The only two things that can trigger react render cycle are either prop updates or state updates. We got the state here to help us.

Reply:
I synced with Martin and Blake offline. For now we will leave this in with a comment above explaining the purpose. It should pose little performance risk as of now.

Reply:
FYI, OpenShift Console has a dedicated [`useForceRender`](https://github.com/openshift/console/blob/master/frontend/packages/console-plugin-shared/src/hooks/useForceRender.ts) hook for this purpose.

Forcing a re-render of the given React component due to a change in certain conditions that happen outside React control flow is not necessarily an anti-pattern.